internal/app/backgroundpublisher.go

```diff
@@ -18,27 +18,40 @@
 package app
 
 import (
+	"fmt"
 	"github.com/edgexfoundry/go-mod-messaging/v2/pkg/types"
 
 	"github.com/edgexfoundry/app-functions-sdk-go/v2/pkg/interfaces"
 )
 
 type backgroundPublisher struct {
-	output chan<- types.MessageEnvelope
+	baseTopic string
```

Comment:
`baseTopic` sounds like it will be added to rather than formatted. I like just `topic`